### ISSUE TITLE
sql: tests in the package time out under race

### ISSUE THREAD

**Issue by yuzefovich** [cockroachdb/cockroach#87307]
[Build](https://teamcity.cockroachdb.com/viewLog.html?buildId=6324778&buildTypeId=Cockroach_BazelExtendedCi)
```
=== RUN   TestVisibilityDuringPrimaryKeyChange
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/b287de7194e17a0386ee1775f0d198ba/logTestVisibilityDuringPrimaryKeyChange1273361120
    test_log_scope.go:79: use -show-logs to present logs inline
panic: test timed out after 59m55s
```

Jira issue: CRDB-19264

**Comment by Xiang-Gu** [comment]
@yuzefovich Hi Yahor, do you know how I might be able reproduce this issue?

I ran this test on Master with `--stress --race` for over an hour and over 3000 runs but still didn't catch it.

**Comment by yuzefovich** [comment]
Interesting. Maybe the problem is that it takes longer than an hour to run all tests in `sql` package together? It seems likely given that there are only two goroutines that have been running for more than minute in the dump.

**Comment by yuzefovich** [comment]
Indeed, I think it's the `sql` package timing out under race - I saw another timeout [here](https://teamcity.cockroachdb.com/viewLog.html?buildId=6367619&buildTypeId=Cockroach_Ci_Tests_Testrace&tab=artifacts#%2Fbazel-testlogs%2Fpkg%2Fsql%2Fsql_test), and it stopped on a different test, after `TestVisibilityDuringPrimaryKeyChange` completed.

**Comment by yuzefovich** [comment]
It seems like it's the time to break up at least the tests in `sql` package, but in the mean time I [asked](https://cockroachlabs.slack.com/archives/C019A8NANH0/p1662521361865349) whether we can increase the timeout.

**Comment by yuzefovich** [comment]
Should be fixed by #87474.
